src/CMakeLists.txt

```diff
@@ -9,8 +9,15 @@
 cuda_rdc_add_library(g4vg SHARED
   G4VG.cc
 )
+# Whilst geocel contains the primary Geant4/VecGeom dependency, we must also 
+# declare the geometry/gdml deps explicitly so that clients can pick them
+# up correctly through LINK_DEPENDENT_LIBRARIES should the linker require it.
+# See: https://discourse.cmake.org/t/use-cases-for-imported-link-dependent-libraries
 cuda_rdc_target_link_libraries(g4vg
-  PRIVATE Celeritas::geocel
+  PRIVATE 
+    Celeritas::geocel 
+    Geant4::G4geometry
+    $<IF:$<TARGET_EXISTS:Geant4::G4gdml>,Geant4::G4gdml,Geant4::G4persistency>
```

Comment:
I'm also curious whether these are needed with the 16 combinations of shared/static for adept/g4vg/celeritas/geant4 🤪

Reply:
I didn't get quite that far....! I'm guessing that they are only needed if Geant4/VecGeom are shared libs. Likely that a shared VecGeom would have similar issues, _but_, the RDC linking needed for that seems to always pull it into the link step anyway...